Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditquerylog/AuditQueryLogImpl.java

```diff
@@ -50,35 +68,80 @@
      *
      * The Adapter logic will be implemented in FHAC-690
      */
-    public QueryAuditEventsResponseType queryAuditEvents(QueryAuditEventsRequestType requestAuditEvents) {
-        return new QueryAuditEventsResponseType();
+    public QueryAuditEventsResponseType queryAuditEvents(QueryAuditEventsRequestType request) {
+        return resultUtil.getQueryAuditEventResponse(getAuditRepositoryDao().queryByAuditOptions(
+            getEventOutcome(request.getEventOutcomeIndicator()), getEventTypes(request.getEventTypeList()),
+            request.getUserId(), getRemoteHcids(request.getRemoteHcidList()), getRequestDate(request.getEventBeginDate()),
+            getRequestDate(request.getEventEndDate())));
+
     }
 
     /**
      *
-     * @param requestAuditEvents - Audit search params MessageId and RelatesTo will be provided by Request. These are
-     * optional fields in requestAuditEvents
+     * @param request - Audit search params MessageId and RelatesTo will be provided by Request. These are optional
+     * fields in requestAuditEvents
      * @return QueryAuditEventsResponseType - The Response will be having EventType or ServiceName, EventStatus- Success
      * or Failure, Event Timestamp, UserId(Human who initiated transaction), Direction (Outbound/Inbound), MessageID
      * -RequestMessageID, RelatesTo (Relates the DeferredRequests and DeferredResponses, Remote Organization Id and
      * Audit Id.
      *
-     * The Adapter logic will be implemented in FHAC-690
      */
     public QueryAuditEventsResponseType queryAuditEventsByMessageId(
-        QueryAuditEventsRequestByRequestMessageId requestAuditEvents) {
-        return new QueryAuditEventsResponseType();
+        QueryAuditEventsRequestByRequestMessageId request) {
```

Comment:
This method does  query by relatesTo id  as well. Can we make the method name more appropriate.